### ISSUE TITLE
unRAID Docker unable to update records

### ISSUE THREAD

**Issue by lukababu** [crazy-max/ddns-route53#224]
### Behaviour
I am not able to update 'A' or 'AAAA' records in route53 after docker installation with unRAID.

#### Steps to reproduce this issue

1. [Create tokens](https://crazymax.dev/ddns-route53/usage/prerequisites/)
2. Create a configuration file using the [official documentation](https://crazymax.dev/ddns-route53/config/)
3. Create initial records in route53 with values "127.0.0.1" for record "A" and "::1" for record "AAAA"
4. Install [ddns-route53](https://forums.unraid.net/profile/92683-a75g/) through [Community Applications plugin](https://forums.unraid.net/topic/38582-plug-in-community-applications/) in unraid
5. add configuration information to /mnt/user/appdata/ddns-route53/ddns-route53.yml using unRAID browser terminal and nano
6. Click apply

#### Expected behaviour

> I expect that the 'A' record for the specified subdomain will have updated values in Route53
> I expect that ipv6 will not be acquired so "AAAA" should not change

#### Actual behaviour

> Neither the 'A' or 'AAAA' record is updating in route53

### Configuration

* ddns-route53 version : crazymax/ddns-route53:latest
* Platform (windows/linux) : unRaid 6.8.3
* System info (type `uname -a`) : Linux unRaid-NA 4.19.107-Unraid #1 SMP Thu Mar 5 13:55:57 PST 2020 x86_64 AMD Ryzen 5 3600X 6-Core Processor AuthenticAMD GNU/Linux

```yml
credentials:
  accessKeyID: "AXXXXXXXXXXXXXXXXXX"
  secretAccessKey: "R+XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"

route53:
  hostedZoneID: "ZXXXXXXXXXXXXXXXXXC"
  recordsSet:
    - name: “unraid-na.example.com.”
      type: "A"
      ttl: 300
    - name: “unraid-na.example.com.”
      type: "AAAA"
      ttl: 300
```

### Logs

```
[90mSun, 06 Sep 2020 15:26:03 PDT[0m [32mINF[0m Starting ddns-route53 [36mversion=[0m2.0.1
[90mSun, 06 Sep 2020 15:26:03 PDT[0m [32mINF[0m Configuration loaded from file: /ddns-route53.yml
[90mSun, 06 Sep 2020 15:26:03 PDT[0m [33mDBG[0m No DDNSR53_* environment variables defined
[90mSun, 06 Sep 2020 15:26:03 PDT[0m [33mDBG[0m {
"credentials": {
"accessKeyID": "AXXXXXXXXXXXXXXXXXX",
"secretAccessKey": "R+XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
},
"route53": {
"hostedZoneID": "ZXXXXXXXXXXXXXXXXXC",
"recordsSet": [
{
"name": "“unraid-na.example.com.”",
"type": "A",
"ttl": 300
},
{
"name": "“unraid-na.example.com.”",
"type": "AAAA",
"ttl": 300
}
]
}
}
[90mSun, 06 Sep 2020 15:26:04 PDT[0m [32mINF[0m Current WAN IPv4: 135.23.XXX.XXX
[90mSun, 06 Sep 2020 15:26:04 PDT[0m [1m[31mERR[0m[0m Cannot retrieve WAN IPv6 address [31merror=[0m[31m"request failed: Get https://v6.ident.me/: dial tcp [2a01:7e00::f03c:91ff:fe70:2b9d]:443: connect: network is unreachable"[0m

[90mSun, 06 Sep 2020 15:26:04 PDT[0m [1m[31mERR[0m[0m No WAN IPv6 address available to update “unraid-na.iremadze.com.” record
[90mSun, 06 Sep 2020 15:26:05 PDT[0m [1m[31mERR[0m[0m Cannot update records set [31merror=[0m[31m"InvalidInput: Invalid XML ; cvc-complex-type.2.4.b: The content of element 'Change' is not complete. One of '{\"https://route53.amazonaws.com/doc/2013-04-01/\":Action, \"https://route53.amazonaws.com/doc/2013-04-01/\":ResourceRecordSet}' is expected.\n\tstatus code: 400, request id: 1c887be7-3c9c-4c9b-8c60-53fc17d26bb6"[0m
[90mSun, 06 Sep 2020 15:26:05 PDT[0m [32mINF[0m 2 records set updated [36mchanges=[0m{"ChangeInfo":null}
[90mSun, 06 Sep 2020 15:26:05 PDT[0m [32mINF[0m Cron initialized with schedule */30 * * * *
[90mSun, 06 Sep 2020 15:26:05 PDT[0m [32mINF[0m Next run in 3 minutes (2020-09-06 15:30:00 -0700 PDT)
```

**Comment by crazy-max** [comment]
@lukababu 

`"name": "“unraid-na.example.com.”",`

As you can see in your logs `name` of your `recordsSet` entries are wrong.

`- name: “unraid-na.example.com.”` should be `- name: "unraid-na.example.com."`

**Comment by lukababu** [comment]
Wow such an annoying thing. Regardless it is not working:

### New Debug Log
```
ErrorWarningSystemArrayLogin


[90mSun, 06 Sep 2020 16:49:42 PDT[0m [32mINF[0m Starting ddns-route53 [36mversion=[0m2.0.1
[90mSun, 06 Sep 2020 16:49:42 PDT[0m [32mINF[0m Configuration loaded from file: /ddns-route53.yml
[90mSun, 06 Sep 2020 16:49:42 PDT[0m [33mDBG[0m No DDNSR53_* environment variables defined
[90mSun, 06 Sep 2020 16:49:42 PDT[0m [33mDBG[0m {
"credentials": {
"accessKeyID": "AXXXXXXXXXXXXXXXXXX",
"secretAccessKey": "R+XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
},
"route53": {
"hostedZoneID": "ZXXXXXXXXXXXXXXXXXC",
"recordsSet": [
{
"name": "unraid-na.example.com.",
"type": "A",
"ttl": 300
},
{
"name": "unraid-na.example.com.",
"type": "AAAA",
"ttl": 300
}
]
}
}
[90mSun, 06 Sep 2020 16:49:43 PDT[0m [32mINF[0m Current WAN IPv4: 135.23.XXX.XXX
[90mSun, 06 Sep 2020 16:49:43 PDT[0m [1m[31mERR[0m[0m Cannot retrieve WAN IPv6 address [31merror=[0m[31m"request failed: Get https://v6.ident.me/: dial tcp [2a01:7e00::f03c:91ff:fe70:2b9d]:443: connect: network is unreachable"[0m

[90mSun, 06 Sep 2020 16:49:43 PDT[0m [1m[31mERR[0m[0m No WAN IPv6 address available to update unraid-na.iremadze.com. record
[90mSun, 06 Sep 2020 16:49:43 PDT[0m [1m[31mERR[0m[0m Cannot update records set [31merror=[0m[31m"InvalidInput: Invalid XML ; cvc-complex-type.2.4.b: The content of element 'Change' is not complete. One of '{\"https://route53.amazonaws.com/doc/2013-04-01/\":Action, \"https://route53.amazonaws.com/doc/2013-04-01/\":ResourceRecordSet}' is expected.\n\tstatus code: 400, request id: 73defa0c-dc1b-409a-b52f-66a36b14ae2e"[0m
[90mSun, 06 Sep 2020 16:49:43 PDT[0m [32mINF[0m 2 records set updated [36mchanges=[0m{"ChangeInfo":null}
[90mSun, 06 Sep 2020 16:49:43 PDT[0m [32mINF[0m Cron initialized with schedule */30 * * * *
[90mSun, 06 Sep 2020 16:49:43 PDT[0m [32mINF[0m Next run in 10 minutes (2020-09-06 17:00:00 -0700 PDT)
```

The part where it says _No DDNSR53_* environment variables defined_ worries me, first time I used AWS IAM so could it be I am passing something wrong?

**Comment by crazy-max** [comment]
@lukababu

> The part where it says No DDNSR53* environment variables defined_ worries me, first time I used AWS IAM so could it be I am passing something wrong?

That's ok if you only use the [YAML configuration file](https://crazymax.dev/ddns-route53/config/) and don't use environment variables.

> Regardless it is not working:

`Cannot retrieve WAN IPv6 address`: IPv6 is disabled on your WAN so you can't update `AAAA` record. Remove this record and it should be ok.

**Comment by lukababu** [comment]
That worked thank you!

My intention was to set this up once and hoped that once my ISP offers ipv6 it would automatically grab it.
I guess a new feature request would be to put in the last known IP or localhost if no IP can be retrieved.

**Comment by crazy-max** [comment]
@lukababu 

> I guess a new feature request would be to put in the last known IP or localhost if no IP can be retrieved.

Yes I will enhanced this part. Thanks for your input and thanks a LOT for sponsoring me :heart:

**Comment by crazy-max** [comment]
@lukababu 

> I guess a new feature request would be to put in the last known IP or localhost if no IP can be retrieved.

It has been implemented trough 2.1.0. Can you try it?

**Comment by lukababu** [comment]
> @lukababu
> 
> > I guess a new feature request would be to put in the last known IP or localhost if no IP can be retrieved.
> 
> It has been implemented trough 2.1.0. Can you try it?

Confirm, it's working with the original configuration. I reset the IP for 'A' and 'AAAA' in Route53 to localhost and it changed 'A' leaving 'AAAA' unaffected.

Thank you!
